### ISSUE TITLE
Error when updating index

### ISSUE THREAD

**Issue by jbellonr** [lucidworks/hive-solr#5]
Hello,

We've got this error when updating the index. We hace defined the following Hive table and can query index data, but not updating it. Any suggestion?

```
CREATE EXTERNAL TABLE hive_solr_test_1_zk (id string, field1 string)
      STORED BY 'com.lucidworks.hadoop.hive.LWStorageHandler'
      LOCATION '/tmp/solr'
      TBLPROPERTIES('solr.zkhost' = 'quickstart.cloudera:2181/solr',
                    'solr.collection' = 'hivesolrtest1',
                    'solr.query' = '*:*');
  ```                  
              
```java
16/05/02 10:14:44 INFO mapreduce.Job: The url to track the job: http://quickstart.cloudera:8088/proxy/application_1461934007258_0005/
16/05/02 10:14:44 INFO exec.Task: Starting Job = job_1461934007258_0005, Tracking URL = http://quickstart.cloudera:8088/proxy/application_1461934007258_0005/
16/05/02 10:14:44 INFO exec.Task: Kill Command = /usr/lib/hadoop/bin/hadoop job  -kill job_1461934007258_0005
16/05/02 10:15:00 INFO exec.Task: Hadoop job information for Stage-0: number of mappers: 1; number of reducers: 0
16/05/02 10:15:00 WARN mapreduce.Counters: Group org.apache.hadoop.mapred.Task$Counter is deprecated. Use org.apache.hadoop.mapreduce.TaskCounter instead
16/05/02 10:15:00 INFO exec.Task: 2016-05-02 10:15:00,436 Stage-0 map = 0%,  reduce = 0%
16/05/02 10:15:44 WARN mapreduce.Counters: Group org.apache.hadoop.mapred.Task$Counter is deprecated. Use org.apache.hadoop.mapreduce.TaskCounter instead
16/05/02 10:15:44 INFO exec.Task: 2016-05-02 10:15:44,311 Stage-0 map = 100%,  reduce = 0%
16/05/02 10:15:45 WARN mapreduce.Counters: Group org.apache.hadoop.mapred.Task$Counter is deprecated. Use org.apache.hadoop.mapreduce.TaskCounter instead
16/05/02 10:15:45 ERROR exec.Task: Ended Job = job_1461934007258_0005 with errors
16/05/02 10:15:45 INFO impl.YarnClientImpl: Killed application application_1461934007258_0005
16/05/02 10:15:45 ERROR ql.Driver: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask
16/05/02 10:15:45 INFO log.PerfLogger: </PERFLOG method=Driver.execute start=1462176883595 end=1462176945435 duration=61840 from=org.apache.hadoop.hive.ql.Driver>
16/05/02 10:15:45 INFO ql.Driver: MapReduce Jobs Launched: 
16/05/02 10:15:45 WARN mapreduce.Counters: Group FileSystemCounters is deprecated. Use org.apache.hadoop.mapreduce.FileSystemCounter instead
16/05/02 10:15:45 INFO ql.Driver: Stage-Stage-0: Map: 1   HDFS Read: 0 HDFS Write: 0 FAIL
16/05/02 10:15:45 INFO ql.Driver: Total MapReduce CPU Time Spent: 0 msec
16/05/02 10:15:45 INFO log.PerfLogger: <PERFLOG method=releaseLocks from=org.apache.hadoop.hive.ql.Driver>
16/05/02 10:15:45 INFO ZooKeeperHiveLockManager:  about to release lock for ocean_tests/hive_solr_test_1_zk2
16/05/02 10:15:45 INFO ZooKeeperHiveLockManager:  about to release lock for ocean_tests/dwh_kunde
16/05/02 10:15:45 INFO ZooKeeperHiveLockManager:  about to release lock for ocean_tests
16/05/02 10:15:45 INFO log.PerfLogger: </PERFLOG method=releaseLocks start=1462176945444 end=1462176945452 duration=8 from=org.apache.hadoop.hive.ql.Driver>
16/05/02 10:15:45 ERROR operation.Operation: Error running hive query: 
org.apache.hive.service.cli.HiveSQLException: Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask
	at org.apache.hive.service.cli.operation.SQLOperation.runInternal(SQLOperation.java:147)
	at org.apache.hive.service.cli.operation.SQLOperation.access$000(SQLOperation.java:69)
	at org.apache.hive.service.cli.operation.SQLOperation$1$1.run(SQLOperation.java:200)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:415)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1642)
	at org.apache.hadoop.hive.shims.HadoopShimsSecure.doAs(HadoopShimsSecure.java:502)
	at org.apache.hive.service.cli.operation.SQLOperation$1.run(SQLOperation.java:213)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by ctargett** [comment]
@jbellonr I noticed you were referencing `quickstart.cloudera:2181/solr` in your solr.zkhost parameter. Does that mean you are trying to index to the Solr included with Cloudera Search? Can you share the versions of Cloudera/Solr you are trying to index to?

**Comment by acesar** [comment]
@jbellonr Can you share your use case?
What command did you use to get the exception?
>  but not updating it
You mean that your are trying to re-index new content?

**Comment by jbellonr** [comment]
Hi Cesar,

We've got two environments (production & development) and it doesn't work at none of both.
Versions are:

production: Cloudera 5.5.0
development: Cloudera 5.5.2

I send you this link where you can see the component version for every Cloudera package.

http://www.cloudera.com/documentation/enterprise/release-notes/topics/cdh_vd_cdh_package_tarball_55.html#concept_wcr_lvz_c5

Keep waiting for good news :)


Thank you.

**Comment by jbellonr** [comment]
Hi again Cesar,

I attach this powerpoint where you can see all about the error tracking.
[error tracking.pptx](https://github.com/lucidworks/hive-solr/files/274487/error.tracking.pptx)



Thank you.

**Comment by acesar** [comment]
@jbellonr The `select` statement seems to be wrong:
```sql
insert into <table>
select '4' 'cuarto'
```
Usually a `select` statement is:
```sql
select <fields> from <table>
```

You can check more info here: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Select

**Comment by jbellonr** [comment]
Hi César,

There is no problem with this select statement. It works perfectly in hive
and impala. You can test in cloudera.

@jbellonr <https://github.com/jbellonr> The select statement seems to be
wrong:

insert into <table>select '4' 'cuarto'

Usually a select statement is:

select <fields> from <table>

You can check more info here:
https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Select

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/lucidworks/hive-solr/issues/5#issuecomment-220645498>

**Comment by acesar** [comment]
It seems to be an issue with the version of hive that you are using.
https://issues.apache.org/jira/browse/HIVE-12200
The problem should be fixed in later versions [hive `1.3.0`,  `2.0.0`], please upgrade and retry.
